core/chains/evm/logpoller/log_poller.go

```diff
@@ -796,26 +801,29 @@ func (lp *logPoller) PollAndSaveLogs(ctx context.Context, currentBlockNumber int
 
 	for {
 		h := currentBlock.Hash
-		var logs []types.Log
-		logs, err = lp.ec.FilterLogs(ctx, lp.Filter(nil, nil, &h))
+		var gethLogs []types.Log
+		gethLogs, err = lp.ec.FilterLogs(ctx, lp.Filter(nil, nil, &h))
 		if err != nil {
 			lp.lggr.Warnw("Unable to query for logs, retrying", "err", err, "block", currentBlockNumber)
 			return
 		}
-		lp.lggr.Debugw("Unfinalized log query", "logs", len(logs), "currentBlockNumber", currentBlockNumber, "blockHash", currentBlock.Hash, "timestamp", currentBlock.Timestamp.Unix())
-		err = lp.orm.q.WithOpts(pg.WithParentCtx(ctx)).Transaction(func(tx pg.Queryer) error {
+		lp.lggr.Debugw("Unfinalized log query", "logs", len(gethLogs), "currentBlockNumber", currentBlockNumber, "blockHash", currentBlock.Hash, "timestamp", currentBlock.Timestamp.Unix())
+		err = lp.saveLogsInTx(ctx, func(tx pg.Queryer) ([]Log, error) {
 			if err2 := lp.orm.InsertBlock(h, currentBlockNumber, currentBlock.Timestamp, pg.WithQueryer(tx)); err2 != nil {
-				return err2
+				return nil, err2
 			}
-			if len(logs) == 0 {
-				return nil
+			if len(gethLogs) == 0 {
+				return nil, nil
```

Comment:
Could this check not be moved up outside of this function closure?

Reply:
If we do that then we won't be inserting a block, is that what we want?

Reply:
Yup missed the block insert.